openwisp_monitoring/monitoring/static/monitoring/js/chart.js

```diff
@@ -16,6 +16,128 @@
         }
         return newArr;
     }
+
+    function adaptiveFilterPoints(charts, yRawVal) {
+
+        var convertToAdaptiveBytes = function(charts, type, i, multiplier, unit) {
+            charts[type].y[i] = Math.round((charts[type].y[i] * multiplier) * 100) / 100;
+            charts[type].hovertemplate[i] = charts[type].y[i] + ' ' + unit;
+        };
+
+        for (var i=0; i<charts[0].y.length; i++) {
```

Comment:
@yashikajotwani12 I think the following code can be reduced by using nested `for` loops. E.g.: 

```js
for (var i=0; i<charts.length; i++) {
    for (var j=0; j<charts[i].y.length; j++) {
          // logic for converting bytes 
   }
}
```

Is there any specific reason that you wrote it in the following manner (does logic of these loops differ in a way which is not immediately visible) ?

Reply:
definitely there's a lot of repetition here, the same code is repeated 3 times @yashikajotwani12!